SYCL/DeviceLib/imf_fp64_test.cpp

```diff
@@ -0,0 +1,276 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+
+namespace s = cl::sycl;
+constexpr s::access::mode sycl_read = s::access::mode::read;
+constexpr s::access::mode sycl_write = s::access::mode::write;
+
+extern "C" {
+float __imf_double2float_rd(double);
+float __imf_double2float_rn(double);
+float __imf_double2float_ru(double);
+float __imf_double2float_rz(double);
+int __imf_double2hiint(double);
+int __imf_double2loint(double);
+int __imf_double2int_rd(double);
+int __imf_double2int_rn(double);
+int __imf_double2int_ru(double);
+int __imf_double2int_rz(double);
+long long int __imf_double2ll_rd(double);
+long long int __imf_double2ll_rn(double);
+long long int __imf_double2ll_ru(double);
+long long int __imf_double2ll_rz(double);
+}
+
+void run_double2float_test(s::queue &queue) {
+  // double xx = 1 + std::numeric_limits<double>::epsilon();
+  bool pass = true;
+  static const size_t NUM = 8;
+  double input_vals[NUM] = {0.0,
+                            2.5,
+                            23.3457746363636,
+                            123877377737.2323233232,
+                            -41.3249382432435,
+                            -6.7543E+25,
+                            9.54E-20,
+                            -3.12E+19};
+  float output_vals[NUM][4];
+  unsigned int refer_vals[NUM][4] = {
+      {0, 0, 0, 0},
+      {0x40200000, 0x40200000, 0x40200000, 0x40200000},
+      {0x41bac425, 0x41bac425, 0x41bac426, 0x41bac425},
+      {0x51e6bd56, 0x51e6bd56, 0x51e6bd57, 0x51e6bd56},
+      {0xc2254cbd, 0xc2254cbd, 0xc2254cbc, 0xc2254cbc},
+      {0xea5f7b26, 0xea5f7b25, 0xea5f7b25, 0xea5f7b25},
+      {0x1fe141c2, 0x1fe141c3, 0x1fe141c3, 0x1fe141c2},
+      {0xdfd87e56, 0xdfd87e55, 0xdfd87e55, 0xdfd87e55}};
```

Comment:
Do we have a native C++ reference implementation of those (i.e. `std::*`) ?

Reply:
Hi, @aelovikov-intel 
There is no native C++ reference implementation for those APIs.

Thanks very much.